### DISCUSSION TITLE
Any option to use older data-weave version?

### DISCUSSION THREAD

**Discussion by krixerx** [mulesoft-labs/data-weave-native#46] [Ideas]
Is there any way to use older version (%dw 1.0) of DataWeave?
Or is there any converter or transformer to convert older schema to new one (%dw 2.0) ?

**Comment by machaval** [comment]
Hi currently we don't have a way to run old DataWeave scripts into this cli it only run DW 2.0 files
And yes there is a tool to migrate mule 3 to mule 4 apps https://github.com/mulesoft/mule-migration-assistant that will translate DW 1 into 2. Though I think it would be a good idea to have a way to migrate dw 1 to into 2 inside the cli. Something like `dw --migrate ./pathToDW1` I will report it and see If I can get move forward

**Reply by machaval** [reply]
Hi @krixerx  The latest version of the cli 1.0.28 has a new option --migrate that expects a DW 1 file and output the DW2 code. This will allow you to migrate your DW1 files and enjoy the new DW2 goodies. Hope you like it and le me know if works for you?
